### PR TITLE
Added automation script for form-builder

### PR DESCRIPTION
**Pull request by priyanshu1602**

### CODE REVIEW

File: tests/admin.js [hunk 1]
```diff
@@ -283,4 +286,88 @@ step("Verify Event <message> in Audit log for the <user>", async function (strMe
...
+    var formName = gauge.dataStore.scenarioStore.get("FormName")
+    await click(link(toRightOf(formName)))
+});
+
+step("Create obs group <arg0> <arg1>", async function (obsName, properties) {
+    await dragAndDrop("ObsGroup", $(".form-builder-row"));
+    await click("Select ObsGroup Source")
+    await write(obsName, into(textBox(below("Control Properties"))))
+    await press('Enter')
+    await click("Coronary Artery Disease, Risk factors")
```
Comment (gokultw) [inline tests/admin.js:306]: why this is hard coded?

File: tests/admin.js [hunk 2]
```diff
@@ -283,4 +286,88 @@ step("Verify Event <message> in Audit log for the <user>", async function (strMe
...
+});
+
+step("Click on Manage Forms", async function () {
+    await click("Manage Forms");
+});
+
+step("Open the form created through form builder", async function () {
+    var formName = gauge.dataStore.scenarioStore.get("FormName")
+    await click($("//a[normalize-space()='" + formName + "']"))
+    await click(formName)
```
Comment (gokultw) [inline tests/admin.js:331]: why clicking twice?

File: tests/admin.js [hunk 3]
```diff
@@ -283,4 +286,88 @@ step("Verify Event <message> in Audit log for the <user>", async function (strMe
...
+});
+
+step("Click on delete form", async function () {
+    confirm('Are you sure you want to delete this entire form AND schema?', async () => await accept());
+    await click($("//input[@value='Delete Form']"))
+});
+
+step("Validate the report generated for Snomed form builder form Report <arg0>", async function (observationFormFile) {
```
Comment (gokultw) [inline tests/admin.js:339]: arg0 arg1 -> update this args name where ever you have used.

File: data/consultation/observations/testForm.json [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                    "short_name": "Is patient on max dose anti-anginal medicines",
+                    "value": "Yes",
+                    "shortValue": "True"
+                },
+                {
+                    "type": "DropDown",
+                    "label": "Coronary Artery Disease, Risk factors",
+                    "short_name": "Risk factors",
+                    "value": "Structure of apex of heart",
+                    "fullName":"Structure of apex of heart (body structure)"
```
Comment (gokultw) [inline data/consultation/observations/testForm.json:59]: fullName?
looks like shortvalue & fullValue!

File: tests/admin.js [hunk 5]
```diff
@@ -283,4 +286,88 @@ step("Verify Event <message> in Audit log for the <user>", async function (strMe
...
+        else if (configuration.type === 'DropDown') {
+            var expected = configuration.fullName
+            var actual = await getActualResult(configuration.short_name)
+            assert.equal(actual, expected);
+        }
+        else {
+            var expected = configuration.value
+            var actual = await getActualResult(configuration.short_name)
+            assert.equal(actual, expected);
+        }
```
Comment (gokultw) [inline tests/admin.js:364]: Many duplications..
Optmize this block

File: tests/admin.js [hunk 6]
```diff
@@ -283,4 +286,88 @@ step("Verify Event <message> in Audit log for the <user>", async function (strMe
...
+    }
+
+}
+async function getActualResult(short_name) {
```
Comment (gokultw) [inline tests/admin.js:368]: getActualResult -> update method name.

**Review by gokultw — CHANGES_REQUESTED** [review]

File: tests/admin.js (hunk 1 above)
Comment (priyanshu1602) [reply tests/admin.js]: Fixed

File: tests/admin.js (hunk 2 above)
Comment (priyanshu1602) [reply tests/admin.js:330]: Removed second one
Fixed

File: tests/admin.js (hunk 3 above)
Comment (priyanshu1602) [reply tests/admin.js:338]: updated

File: data/consultation/observations/testForm.json (hunk 4 above)
Comment (priyanshu1602) [reply data/consultation/observations/testForm.json]: Fixed

File: tests/admin.js (hunk 5 above)
Comment (priyanshu1602) [reply tests/admin.js]: Fixed

File: tests/admin.js (hunk 6 above)
Comment (priyanshu1602) [reply tests/admin.js]: Fixed

**Pull request merged**